lib/maths/CDataFrameCategoryEncoder.cc

```diff
@@ -190,169 +211,396 @@ class CMinRedundancyMaxRelevancyGreedySearch {
+                core::CPersistUtils::restore(ENCODING_MIC_TAG, m_Mic, traverser))
+        if (this->acceptRestoreTraverserSubroutine(traverser) == false) {
+            return false;
+        }
 
-    this->setupTargetMeanValueEncoding(parameters.m_NumberThreads, parameters.m_Frame,
-                                       parameters.m_RowMask, categoricalColumnMask,
-                                       parameters.m_TargetColumn);
+    } while (traverser.next());
+    return true;
+}
 
-    this->finishEncoding(
-        parameters.m_TargetColumn,
-        this->selectFeatures(parameters.m_NumberThreads, parameters.m_Frame,
-                             parameters.m_RowMask, metricColumnMask,
-                             categoricalColumnMask, parameters.m_TargetColumn));
+CDataFrameCategoryEncoder::CIdentityEncoding::CIdentityEncoding(std::size_t inputColumnIndex,
+                                                                double mic)
+    : CEncoding{inputColumnIndex, mic} {
 }
 
-CDataFrameCategoryEncoder::CDataFrameCategoryEncoder(core::CStateRestoreTraverser& traverser) {
-    if (traverser.traverseSubLevel(std::bind(&CDataFrameCategoryEncoder::acceptRestoreTraverser,
-                                             this, std::placeholders::_1)) == false) {
-        throw std::runtime_error{"failed to restore category encoder"};
-    }
+EEncoding CDataFrameCategoryEncoder::CIdentityEncoding::type() const {
+    return E_IdentityEncoding;
 }
 
-CEncodedDataFrameRowRef CDataFrameCategoryEncoder::encode(TRowRef row) const {
-    return {std::move(row), *this};
+double CDataFrameCategoryEncoder::CIdentityEncoding::encode(double value) const {
+    return value;
 }
 
-bool CDataFrameCategoryEncoder::columnIsCategorical(std::size_t feature) const {
-    return m_ColumnIsCategorical[feature];
+bool CDataFrameCategoryEncoder::CIdentityEncoding::isBinary() const {
+    return false;
 }
 
-const CDataFrameCategoryEncoder::TDoubleVec& CDataFrameCategoryEncoder::featureMics() const {
-    return m_FeatureVectorMics;
+std::uint64_t CDataFrameCategoryEncoder::CIdentityEncoding::checksum() const {
+    return CChecksum::calculate(this->inputColumnIndex(), this->mic());
 }
 
-std::size_t CDataFrameCategoryEncoder::numberFeatures() const {
-    return m_FeatureVectorMics.size();
+std::string CDataFrameCategoryEncoder::CIdentityEncoding::typeString() const {
+    return IDENTITY_ENCODING_TAG;
 }
 
-std::size_t CDataFrameCategoryEncoder::encoding(std::size_t index) const {
-    return m_FeatureVectorEncodingMap[index];
+void CDataFrameCategoryEncoder::CIdentityEncoding::acceptPersistInserterSubroutine(
+    core::CStatePersistInserter& inserter) const {
+    // do nothing
 }
 
-std::size_t CDataFrameCategoryEncoder::column(std::size_t index) const {
-    return m_FeatureVectorColumnMap[index];
+bool CDataFrameCategoryEncoder::CIdentityEncoding::acceptRestoreTraverserSubroutine(
+    core::CStateRestoreTraverser& traverser) {
```

Comment:
This unused parameters will generate a warning
```suggestion
    core::CStateRestoreTraverser& /*traverser*/) {
```

Reply:
Done